<launch>

  <!-- ARGUMENTS -->
  <arg name="use_joy" default="false"            />
  <arg name="joy_dev" default="/dev/input/js0"  />

  <!--
  DRIVERS
  -->

  <!-- Cozmo driver -->
  <node ns="cozmo1"
        pkg="cozmo_driver" type="cozmo_driver.py" name="cozmo_driver_node"
        output="screen" >
    <param name="serial_port" value="04e0d4110a65c4ee" />
    <!-- <param name="serial_port" value="ZY223BMTVP" /> -->
    <param name="odom_frame" value="cozmo1/odom" />
    <param name="footprint_frame" value="cozmo1/base_footprint" />
    <param name="base_frame" value="cozmo1/base_link" />
    <param name="head_frame" value="cozmo1/head_link" />
    <param name="camera_frame" value="cozmo1/camera_link" />
    <param name="camera_info_url" value="file://$(find cozmo_driver)/config/cozmo_camera.yaml" />
  </node>

  <node ns="cozmo2"
        pkg="cozmo_driver" type="cozmo_driver.py" name="cozmo_driver_node"
        output="screen" >
    <!-- <param name="serial_port" value="04e0d4110a65c4ee" /> -->
    <param name="serial_port" value="ZY223BMTVP" />
    <param name="odom_frame" value="cozmo2/odom" />
    <param name="footprint_frame" value="cozmo2/base_footprint" />
    <param name="base_frame" value="cozmo2/base_link" />
    <param name="head_frame" value="cozmo2/head_link" />
    <param name="camera_frame" value="cozmo2/camera_link" />

    <param name="camera_info_url" value="file://$(find cozmo_driver)/config/cozmo_camera.yaml" />
  </node>



  <!-- Head angle / lift height teleoperation -->
  <!--<node pkg="cozmo_driver" type="head_lift_joy.py" name="head_lift_joy_node"
        output="screen" > -->
    <!-- <remap from="joy"         to="joy_teleop/joy"     /> -->
    <!-- <remap from="cmd_vel"         to="cozmo/cmd_vel"     /> -->
    <!--  <remap from="head_angle"  to="cozmo/head_angle"   />
    <remap from="lift_height" to="cozmo/lift_height"  />
  </node>-->




  <!--
  INPUT
  -->

  <!-- Control -->
<!--   <node pkg="twist_mux" type="twist_mux" name="twist_mux_node">
    <rosparam command="load" file="$(find cozmo_driver)/config/twist_mux.yaml" />
    <remap from="cmd_vel_out" to="cozmo/cmd_vel"/>
  </node> -->

  <!-- Teleoperation -->
  <group if="$(arg use_joy)">
    <rosparam command="load" file="$(find cozmo_driver)/config/teleop.yaml" />

    <node pkg="joy" type="joy_node" name="joy_node" >
      <param name="dev" value="$(arg joy_dev)" />
    </node>

    <node pkg="teleop_twist_joy" type="teleop_node" name="teleop_node">
      <remap from="cmd_vel"         to="cozmo/cmd_vel"     />
    </node>
  </group>

<!--   <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop_twist_keyboard">
      <remap from="cmd_vel"         to="cozmo/cmd_vel"     />
  </node> -->

  <!--
  VISUALIZATION
  -->
  <node pkg="map_server" type="map_server" name="map_server" args="$(find cozmo_driver)/maps/cozmo.yaml" />

  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find cozmo_driver)/config/cozmo.rviz"
        output="screen" />

</launch>
